### PR TITLE
fit with aspect ratios

### PR DESCRIPTION
**Pull request by wpalmer**
I've made a couple of changes to allow "fit" to optionally preserve aspect-ratio.

I did these changes in "cycle.lite.1.1.js", then ported them over to cycle.all since I couldn't find a branch for just the "lite" part. The makes it highly likely that there are bugs here, since I didn't bother to thoroughly test the .all version.

It also "crops when necessary", which is fine for my needs, but may not be universally desired.

In summary: this probably needs work prior to inclusion. I'm willing to do that extra work if necessary, but I figured a pull request was a good way to find out 1) if it really is necessary, and 2) if there's any chance of getting this feature added in the first place.

I release these changes under the MIT and GPL licenses as required.

### CODE REVIEW

**Comment by malsup** [comment]
Merged.  Thanks!
